src/main/java/com/google/sps/servlets/PromotionQuizServlet.java

```diff
@@ -60,18 +73,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    this.quizQuestions = this.questionDatabase.getQuizQuestions();
-    Boolean isPromoted = handleQuizSubmission(request);
-    PromotionMessage promotionMessage =
-        new PromotionMessage(isPromoted, isPromoted ? PROMOTED_MESSAGE : NOT_PROMOTED_MESSAGE);
-    if (isPromoted) {
-      // TODO: https://github.com/googleinterns/step144-2020/issues/88
-      // then get the next id by calling currentGameStage.getNextStageId()
-      // then set the players gameStage to the next id
+    try {
+      this.quizQuestions = getQuizQuestions(response);
+      Boolean isPromoted = handleQuizSubmission(request);
+      PromotionMessage promotionMessage =
+          new PromotionMessage(isPromoted, isPromoted ? PROMOTED_MESSAGE : NOT_PROMOTED_MESSAGE);
+      if (isPromoted) {
```

Comment:
What does the user do if they're not promoted? Do they redo the quiz? Do they go through the content again?

Reply:
Their game stage will be unchanged, and they will be redirected back to the game stage page to go through the content again. In a future PR, I will be linking the promotion quiz functionality to the game stage template.